### PR TITLE
New features and fixes for the autopwn command

### PR DESCRIPTION
**Pull request by matthiaskonrath**
Add feature to extract the B keys with known A keys (I couldn't verify if this is working correct, so someone should please try it first)
Add some of my found keys in the dictionary
Add a safer function to load dictionaries from the filesystem
Add fast key check for nested keys (I have no idea why this works there and not in the key reuse detection loop above...)

### CODE REVIEW

File: client/fileutils.c [hunk 1]
```diff
@@ -649,6 +649,78 @@ int loadFileDICTIONARY(const char *preferredName, void *data, size_t *datalen, u
     return retval;
 }
 
+uint8_t* loadFileDICTIONARY_safe(const char *preferredName, uint8_t keylen, uint16_t *keycnt) {
```
Comment (doegox) [inline client/fileutils.c]: better to return success codes as for loadFileDICTIONARY and to return the pointer to allocated space via pointer of pointer arg

File: client/dictionaries/mfc_default_keys.dic [hunk 2]
```diff
@@ -982,3 +982,36 @@ ef4c5a7ac6fc
...
+# Unknown, AT
+#
+23d4cdff8da3
+e6849fcc324b
+12fd3a94df0e
+#
+# Unknown, AT
+#
+0b83797a9c64
+39ad2963d3d1
```
Comment (doegox) [inline client/dictionaries/mfc_default_keys.dic]: missing \n

**Comment by doegox** [comment]
> Add feature to extract the B keys with known A keys (I couldn't verify if this is working correct, so someone should please try it first)

Yes it works!

**Comment by matthiaskonrath** [comment]
> > Add feature to extract the B keys with known A keys (I couldn't verify if this is working correct, so someone should please try it first)
> 
> Yes it works!

Nice ;)

**Comment by iceman1001** [comment]
hold on,  I have a rewrite of the autopwn...   You will need to merge against that one.. this pr will get conflict.

**Comment by matthiaskonrath** [comment]
> hold on, I have a rewrite of the autopwn... You will need to merge against that one.. this pr will get conflict.

Could you include the feature to extract the B keys with known A keys in your rewrite.
I will then re implement the other stuff in a separate pull request, when you are done.

File: client/fileutils.c (hunk 1 above)
Comment (matthiaskonrath) [reply client/fileutils.c]: with the next pull request, when the rewrite from @iceman1001 is done.

**Comment by iceman1001** [comment]
I just merged it.  So.. yeah.  How about you split things up.   Like the fileutils change,  (don't return pointer like @doegox  said,  use **)   do that in a separate PR.    And then we can use that as base for the rest of the changes in the files that uses loadDictionary..    However,  don't touch cmdhficlass / iclass stuff.  I am working on it.

**Comment by doegox** [comment]
@matthiaskonrath be careful I pushed some style fix just after iceman, better to branch from there

**Comment by doegox** [comment]
> Could you include the feature to extract the B keys with known A keys in your rewrite.

Still to be done again I'm afraid

**Pull request closed without merging**